### PR TITLE
Modify base image and docker installation scripts to accommodate Ubuntu 22.04 image building

### PR DESCRIPTION
**Pull request by kashifest**
- sudo apt-get upgrade -f -y restarts the services and image building hangs so modifying needrestart  
-  docker-ce docker-ce-cli doesnt have the recommended version anymore for 22.04 so installing the latest. 

Co-authored-by: Lennart Jern lennart.jern@est.tech

### CODE REVIEW

**Review by smoshiur1237 — DISMISSED** [review]

**Review by lentzi90 — CHANGES_REQUESTED** [review]
This won't help unfortunately. It will hang when installing other things, plus we will miss the latest updates.

The reason it hangs is that `needrestart` is included in 22.04. See [this](https://github.com/Nordix/metal3-dev-tools/blob/d514ee69cf62db2e0e1eb67b3518ac7e016efbba/ci/scripts/image_scripts/provision_base_image.sh#L10-L14) from my PR here: https://github.com/Nordix/metal3-dev-tools/pull/536/files

**Comment by kashifest** [comment]
> This won't help unfortunately. It will hang when installing other things, plus we will miss the latest updates.
> 
> The reason it hangs is that `needrestart` is included in 22.04. See [this](https://github.com/Nordix/metal3-dev-tools/blob/d514ee69cf62db2e0e1eb67b3518ac7e016efbba/ci/scripts/image_scripts/provision_base_image.sh#L10-L14) from my PR here: https://github.com/Nordix/metal3-dev-tools/pull/536/files

Good point , Can I do those changes in my PR, I need them now.

**Comment by lentzi90** [comment]
> Good point , Can I do those changes in my PR, I need them now.

Yes, please go ahead! It makes more sense here actually. I'll just rebase then when this is merged :slightly_smiling_face:

**Comment by kashifest** [comment]
Updated @lentzi90

File: ci/scripts/image_scripts/provision_base_image.sh [hunk 1]
```diff
@@ -6,9 +6,15 @@ set -eux
...
+# Needrestart and packer does not seem to work well together. Needrestart is
+# propmpting for what services to restart and packer cannot answer, so it get stuck.
+# This makes needrestart (l)ist the packages instead of prompting with a dialog.
+# The alternative would be sudo apt-get remove -y needrestart.
+echo '$nrconf{restart} = "l";' | sudo tee /etc/needrestart/needrestart.conf || true
+
 # Upgrade all packages
 sudo apt-get update
-sudo apt-get upgrade -f -y
+sudo apt-get dist-upgrade -f -yp
```
Comment (lentzi90) [inline ci/scripts/image_scripts/provision_base_image.sh:17]: What is `-p`? I'm not familiar with this, maybe typo?

**Review by lentzi90 — COMMENTED** [review]
Looks good, just one potential typo below

File: ci/scripts/image_scripts/provision_base_image.sh (hunk 1 above)
Comment (kashifest) [reply ci/scripts/image_scripts/provision_base_image.sh]: typo

**Review by lentzi90 — APPROVED** [review]

**Pull request merged**